internal/state/dataplane/configuration.go

```diff
@@ -11,7 +11,11 @@ import (
 	"github.com/nginxinc/nginx-kubernetes-gateway/internal/state/resolver"
 )
 
-const wildcardHostname = "~^"
+const (
+	wildcardHostname    = "~^"
+	PathMatchPathPrefix = string(v1beta1.PathMatchPathPrefix)
```

Comment:
I think this would be better as a typed string. 

```
type PathType string

const (
    PathTypePrefix = "prefix"
    PathTypeExact = "exact"
)

```

It gives us type safety, removes the need to convert to `v1beta1.PathMatchType` in `nginx/config`, and allows for re-use if we were to handle other resources like ingress.

Reply:
I would still have to perform a conversion from v1beta1 to this type when creating the path name.

Reply:
Actually, I could convert in `dataplane/configuration.go`, right? And then by the time it gets to building the servers, it's already converted.

Reply:
> Actually, I could convert in dataplane/configuration.go, right? And then by the time it gets to building the servers, it's already converted.
Ya, that's what I was thinking